File: java/linkList/linkList.java
// A simple java program to introduce a linked list
class LinkedList {
    Node head;     // head of link list
    /* Linked list node. This inner class is made so that
       main() can access it */

    static class Node {
        int data;
        Node next;
        Node (int d)
        {
            data = d;
            next = null;
        } // Constructor
    }

    /* This function prints contents of linked list starting from head */
    public void printList()
    {
         Node n = head;
         while(n!= null) {
             System.out.print(n.data + " ");
             n = n.next;
         }

    }    

    /* method to create a simple linked list with 3 nodes */
    public static void main(String[] args)
    {
        /* Start with the empty list */
        LinkedList llist = new LinkedList();

        llist.head = new Node(1);
        Node second = new Node(2);
        Node third = new Node(3);

        llist.head.next = second; // link the first node to second node

        second.next = third; // link the second node to third node 
  
        llist.printList();
    }

}

File: python/linkList/linkList.py
#!/bin/python3
# example for simple link List

class Node:

    def __init__(self, data):
        self.data = data  # assign data
        self.next = None  # initialize next as null
 
# Linked List class contains a Node object
class LinkedList:

    # Function to initialize head
    def __init__(self):
        self.head = None

    def printList(self):
        temp = self.head
        while (temp):
             print (temp.data),
             temp = temp.next

if __name__ == '__main__':

   llist = LinkedList()

   llist.head = Node(1)
   second     = Node(2)
   third      = Node(3)
   fourth     = Node(4)

   llist.head.next = second;
   second.next     = third;
   third.next      = fourth;

   llist.printList()


